src/solvers/lowering/byte_operators.cpp

```diff
@@ -18,13 +18,95 @@ Author: Daniel Kroening, kroening@kroening.com
 
 #include "flatten_byte_extract_exceptions.h"
 
-// clang-format off
+static exprt unpack_rec(
+  const exprt &src,
+  bool little_endian,
+  const exprt &max_bytes,
+  const namespacet &ns,
+  bool unpack_byte_array = false);
+
+/// rewrite an array or vector into its individual bytes
+/// \param src: array/vector to unpack
+/// \param src_size: array/vector size
+/// \param element_width: bit width of array/vector elements
+/// \param little_endian: true, iff assumed endianness is little-endian
+/// \param max_bytes: if not nil, use as upper bound of the number of bytes to
+///                   unpack
+/// \param ns: namespace for type lookups
+/// \param [out] dest_array: target array_exprt to hold unpacked elements
+static void unpack_array_vector(
```

Comment:
AFAICT `dest_array` is only non-empty on recursion, in which case split this into a `_rec` with an out-append parameter and a root function that returns an `array_exprt`

Reply:
Done, this makes things a lot clearer I think.